### PR TITLE
Supermassive schema extraction plugin

### PR DESCRIPTION
**Pull request by vejrj**

### CODE REVIEW

File: packages/graphql-codegen-supermassive-schema-extraction-plugin/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { extname } from "path";
+import { RawClientSideBasePluginConfig } from "@graphql-codegen/visitor-plugin-common";
+
+type PluginConfig = RawClientSideBasePluginConfig & {
+  files: string[];
+};
+
+export const plugin: PluginFunction<PluginConfig> = (schema: GraphQLSchema) => {
+  const tsContents: ts.SourceFile = extractImplicitTypesToTypescript(
+    parse(printSchema(schema)),
```
Comment (alloy) [inline packages/graphql-codegen-supermassive-schema-extraction-plugin/src/index.ts:19]: Why do this?

File: packages/graphql-codegen-supermassive-schema-extraction-plugin/CHANGELOG.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            "author": "beachball",
+            "package": "@graphitation/graphql-codegen-supermassive-typed-document-node-plugin",
+            "comment": "Bump @graphitation/supermassive to v0.8.3",
+            "commit": "2f8bfe127192958c94261fd1e6ba998247204a31"
+          }
+        ]
+      }
+    }
+  ]
+}
```
Comment (alloy) [inline packages/graphql-codegen-supermassive-schema-extraction-plugin/CHANGELOG.json:71]: Did you copy this file from the other plugin?

File: packages/graphql-codegen-supermassive-schema-extraction-plugin/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+- No-op bump (eloy.de.enige@gmail.com)
+
+## 0.1.0
+
+Fri, 10 Jun 2022 14:09:50 GMT
+
+### Minor changes
+
+- Supermassive typed document node plugin added (jakubvejr@microsoft.com)
+- Bump @graphitation/supermassive to v0.8.3
```
Comment (alloy) [inline packages/graphql-codegen-supermassive-schema-extraction-plugin/CHANGELOG.md:30]: Ditto

File: packages/graphql-codegen-supermassive-schema-extraction-plugin/CHANGELOG.json (hunk 2 above)
Comment (vejrj) [reply packages/graphql-codegen-supermassive-schema-extraction-plugin/CHANGELOG.json]: Yes, fixed

File: packages/graphql-codegen-supermassive-schema-extraction-plugin/CHANGELOG.md (hunk 3 above)
Comment (vejrj) [reply packages/graphql-codegen-supermassive-schema-extraction-plugin/CHANGELOG.md]: fixed

File: packages/graphql-codegen-supermassive-schema-extraction-plugin/src/index.ts (hunk 1 above)
Comment (vejrj) [reply packages/graphql-codegen-supermassive-schema-extraction-plugin/src/index.ts:19]: extractImplicitTypesToTypescript needs schemaAst not the object

Comment (alloy) [reply packages/graphql-codegen-supermassive-schema-extraction-plugin/src/index.ts:19]: Doesn't this param, or one of the others, already contain the AST input? https://github.com/dotansimha/graphql-code-generator/blob/e3af1dbc80a998444a1d1b353f127f297c6a09d0/packages/utils/plugins-helpers/src/types.ts#L531

Comment (vejrj) [reply packages/graphql-codegen-supermassive-schema-extraction-plugin/src/index.ts:19]: Nope, I don't use the documents field. It's empty array in this case

**Review by alloy — APPROVED** [review]

**Pull request merged**